version: '3.9'

services:
  application:
    build:
      context: .
      dockerfile: application/Dockerfile
    image: chatgpt/application
    container_name: chatgpt-application
    ports:
      - '5001:5001'
    env_file: application/openai.env
    environment:
      TIMEOUT_MS: 60000
      MONGO_URI: mongodb://mongo:27017/chatgpt
    links:
      - mongo
    depends_on:
      - mongo
    volumes:
      - /var/data/chatgpt/log:/var/logs
    command: [ "gunicorn", "-c", "gunicorn_conf.py", "server:app" ]

  nginx:
    build:
      context: .
      dockerfile: nginx/Dockerfile
    image: chatgpt/nginx
    container_name: chatgpt-nginx
    ports:
      - '80:80'
    expose:
      - '80'
    volumes:
      - ./nginx/nginx.conf:/etc/nginx/conf.d/default.conf
      - /var/data/chatgpt/log:/var/log/nginx
      - /var/data/chatgpt/cert:/etc/nginx/cert
    links:
      - application
    depends_on:
      - application
      - mongo
    command: [ "nginx", "-g", "daemon off;" ]

  mongo:
    image: mongo:6
    container_name: chatgpt-mongo
    ports:
      - 27017:27017
    volumes:
      - mongodb_data_container:/data/db

volumes:
  mongodb_data_container: { }
  app_data_container: { }